libregexp.c

```diff
@@ -1201,6 +1201,7 @@ static int find_group_name(REParseState *s, const char *name)
 
     name_len = strlen(name);
     p = (char *)s->group_names.buf;
+    if (!p) return -1;
     buf_end = (char *)s->group_names.buf + s->group_names.size;
```

Comment:
Let's move the strlen call to after the guard but otherwise LGTM (and glaringly obvious bug once you see it so thanks for pointing that out :-))
```suggestion
    p = (char *)s->group_names.buf;
    if (!p) return -1;
    buf_end = (char *)s->group_names.buf + s->group_names.size;
    name_len = strlen(name);
```